### ISSUE TITLE
Error:  .pathUnit_speed

### ISSUE THREAD

**Issue by LuccasRP21** [VictorPhilipp/Cities-Skylines-Traffic-Manager-President-Edition#221]
Priss, 
         The error starts when you start the simulation of cities, whether with all mods or none, as long as you have the traffic manager.

       I'm your fan Victor Philipp, I'm Brazilian.
        There is love to project. I understand. 

        Update 1.10.1 f3.

**Comment by N0spHeratUs** [comment]
Exactly same problem for me

**Comment by N0spHeratUs** [comment]
[output_log.txt](https://github.com/VictorPhilipp/Cities-Skylines-Traffic-Manager-President-Edition/files/2299866/output_log.txt)
[TMPE.log](https://github.com/VictorPhilipp/Cities-Skylines-Traffic-Manager-President-Edition/files/2299867/TMPE.log)

**Comment by LuccasRP21** [comment]
It does not work. I've already done the steps, it keeps giving the error message. I have already copied the saved files, duplicated the file, tried to open it and nothing ..

**Comment by Xyrhenix** [comment]
LuccasRP21: This mod is designed to work after being subscribed to from the Steam Workshop, and is designed to work with the Steam version of Cities: Skylines. Usage of the mod which either 1. has not been downloaded from Steam Workshop, or 2. isn't being used with an official version of Skylines downloaded from the Steam client, is likely to result in bugs. We cannot verify that the versions of the mod or the game are what this mod is intended to work with and bug reports are extremely likely to be misleading and invalid.

In short: We are not going to support piracy.
1. Remove the link from your comment or delete it.
2. Buy the game, subscribe to the mod, and report bugs if there are bugs.
3. Don't report bugs unless you are using official and up-to-date versions of this mod or the game. It's a waste of time for everyone including yourself. Don't complain if you're not using the game from its official source, you should expect bugs if you are going to pirate a game.

**Comment by N0spHeratUs** [comment]
Back from holiday.
I've readed all your answers, so i tried the game on another pc and the mod worked normally. 
So i've checked again, checked the integrity game from steam, unsubscribed and subscribed again the mod, removed all files and directories related to the mod in steam workshop directory, but nothing to do. It was like that steam didn't updated correctly the file games of the mod in steam workshop directory (and i don't know why these files are also located in %localappdata%\Colossal Order\Cities_Skylines\Addons\Mods) . Considering that my steam installation is very very old (like windows, with all updates from windows 7 to 10, so... like 7 years), I've reinstalled from scratch Steam and downloaded from scratch the entire game, subscribed the mod and now it works normally.

**Comment by GalacticRuler** [comment]
Xtrhenix
Unfortunately, not _everyone_ has the disposable income to spend on purchasing a **massively overpriced** game. For those of us who do have to use unofficial copies of the game, which keep in mind _ARE_ up to date usually, a mod as required as this one would be wonderful to have working. In the future, please do refrain from scolding someone who probably would have to choose between food and the game, it brings them joy and they just wanted to seek even an imperfect fix for a broken mod that makes the game actually playable.

I too get this error, for the record; and As it _is_ an unofficial copy of the game, that is up to date with this mod's requirements, not having TMPE work is a bit unfortunate, given I and many other cannot run the mod from the currently required location. It is interesting though, that it cannot be ran from outside the workshop directories, and error logs seem to not help at all in showing why.  No other mod seems to have this problem; granted, most aren't half as complex as this one.

**Comment by Xyrhenix** [comment]
N0spHeratUs: Welcome back from your holiday, and I'm glad that you were able to resolve this issue. As much an inconvenience as it is to reinstall Steam, I am glad it was able to resolve the issue. I'd guess maybe the mod didn't update properly from Steam Workshop.

GalacticRuler: Let me explain in clear terms why mod support for unofficial and illegal copies of the game is not only impractical but nearly impossible.

**There is no way to verify that the copy of the game downloaded from the unofficial / illegal source is functionally the same as that from an official source (minus whatever verification features needed to be bypassed to make it work.** With this in mind, we cannot possibly determine whether it is a problem in this mod's source code or whether it is a problem with the unofficial / illegal copy of the game which is causing the bug which Luccas has. As a result, it will be extremely unlikely that we can fix it unless Luccas uploads for us the exact version of the game he is using so it can be debugged by those capable.

I agree that the mod not working properly from a non-Workshop location is an inconvenience (as I found out through my own testing of the PT fix some time ago) and it should, yet there are no lines of code that I can find which specifically modify behaviour when the mod is used from any specific directory. I can't find a reason for it.

Given that Luccas has informed us that his game may be otherwise modified beyond the APIs and classes that this mod was coded to work with, and that there is no way to guarentee that the game he is using is the same as the one this was coded with, I'd be inclined to consider his bug report invalid. Using the game outside of official expectations (aka buying it) can result in weird and unexpected bugs.

**Comment by GalacticRuler** [comment]
99% of pirate copies of most games, are identical to retail versions of those games. Old myths and rumors about pirate software are very false, excluding a couple of examples that generally are easy to spot and avoid. Luccas explained that his copy shows the same version number as the one required by the mod (and mine does as well), and all other content works exactly as it should; including when it should be broken, it tends to be so.

interesting to know there is in fact no code revolving around where you put the mod. This makes me wonder if it has something to do with where the original version was compiled to? I'm not versed in modding with .dll files however.

**Comment by Xyrhenix** [comment]
It's not necessarily rumours, but more logic. The game has been modified from its original, and it's possible that those modifications can cause bugs. It's useless trying to debug unless it can be verified that all game functionality has been untouched.

> interesting to know there is in fact no code revolving around where you put the mod. This makes me wonder if it has something to do with where the original version was compiled to?

In the source code I noticed compile paths for both Workshop directory and `%LOCALAPPDATA`. See `PostBuildEvent` of TLM/TLM/TLM.csproj. From that, I'd assume testing is done in both directories.

My theory regarding this is that it's not necessarily the mod location causing problems, but rather which location the mod was running from when you saved the game. I think that if you make a save using a version of the mod from the Workshop directory, it'll only work with that specific version. This could be due to the way CSL implements mods (and perhaps, while the mod itself is the same, internally the different paths mean they are different mods to the game) or maybe there is a specific thing done in save.

Still, the theory doesn't make too much sense because then save-sharing could potentially break saves all the time. People have their workshop directory in different places all the time, so it wouldn't make too much sense unless Skylines itself recognises the Steam Workshop directory as a constant variable and adjusts itself accordingly when loading saves.

**Comment by LuccasRP21** [comment]
Sorry for being hacked. But as the colleague said is a value that I have no conditions to have. I would love to officially have it, why would I like to be part of the steam community, report bugs, suggest edits, and be grateful to the effort that programmers have. Unfortunately the game here in BRAZIL =
 250 reais = 70 dollars, with all DLC's. Minimum wage per month = 985 - 250 = 735. We have, dispezas. That's why I live with parents. If I lived alone I would not be able to afford it. Working 8hrs a day. I would have to work 12 hours to get my own support, and join the whole year to buy the game.

I'm sorry, because this my country of MERDA, is so robbery and corrupt. So high taxes.

Thanks to the developer and the German who started this project.

Regarding the game, I downloaded a version 1.10.0 f3 then with errors. When they sent the new version, there was no pirate to download. Until I found this site in version 1.10.1 f3 that lets the TRAFFIC MANAGER mod work. It has a network of pirate mod's as well.

I can not play online, for example. And do not even sign Steam. This hurts me.

Good night to you,
                             18:40 p.m.

**Comment by duy0699cat** [comment]
>be me
>happily bought the game when it on sale and steam start support your country's currency. Bye bye private life. 
>go to military training, shitty stone age network connection 
>cannot connect to steam, start in offline mode
>f*ck the save up then realize no steam mod available 
>after waiting for haft an hour for google loading searches, find a workaround : move mods to game directory 
>load the game, tm:pe insta crash everything 
>spent other 30 mins to search and load to this page

Ok, enough rant, i will just play again when return home. Now i just find something else to do. Always tks for the mod.
